polling failes when document is bound to process on version <= 8.8

__Describe the Bug__

create bpmn with polling and another element with document handling causes polling to fail with error:

```
Error binding properties for HTTP request
java.lang.RuntimeException: com.fasterxml.jackson.databind.JsonMappingException: No serializer found for class io.camunda.connector.runtime.core.document.CamundaDocument and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.HashMap["res"]->java.util.LinkedHashMap["document"]) (through reference chain: io.camunda.connector.http.polling.model.PollingRuntimeProperties["method"])
	at io.camunda.connector.runtime.core.inbound.DefaultProcessInstanceContext.bind(DefaultProcessInstanceContext.java:77)
	at io.camunda.connector.http.polling.task.HttpRequestTask.run(HttpRequestTask.java:41)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No serializer found for class io.camunda.connector.runtime.core.document.CamundaDocument and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.HashMap["res"]->java.util.LinkedHashMap["document"]) (through reference chain: io.camunda.connector.http.polling.model.PollingRuntimeProperties["method"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:400)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:359)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1966)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:304)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:169)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2148)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1755)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1770)
	at io.camunda.connector.runtime.core.inbound.DefaultProcessInstanceContext.bind(DefaultProcessInstanceContext.java:73)
	... 7 more
Caused by: java.lang.IllegalArgumentException: No serializer found for class io.camunda.connector.runtime.core.document.CamundaDocument and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.HashMap["res"]->java.util.LinkedHashMap["document"])
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3686)
	at io.camunda.connector.feel.jackson.AbstractFeelDeserializer.deserialize(AbstractFeelDeserializer.java:66)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:302)
	... 13 more
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class io.camunda.connector.runtime.core.document.CamundaDocument and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.HashMap["res"]->java.util.LinkedHashMap["document"])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1359)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:415)
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.failForEmpty(UnknownSerializer.java:52)
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.serialize(UnknownSerializer.java:29)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeFields(MapSerializer.java:807)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeWithoutTypeInfo(MapSerializer.java:763)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:719)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:34)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeFields(MapSerializer.java:807)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeWithoutTypeInfo(MapSerializer.java:763)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:719)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:34)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3681)
	... 16 more
```

[REST.bpmn.txt](https://github.com/user-attachments/files/26679980/REST.bpmn.txt)

__Steps to Reproduce__

1. do this
2. do that


__Expected Behavior__

<!-- A clear and concise description of what you expected to happen. -->


__Environment__

- OS: [e.g. Windows 7]
- Library version: [e.g. 2.0.0]
- Camunda version: [e.g. 8.0.4 (self-managed)] --> please also use affects/8.X label

--- a/connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/document/BlankObjectMapperDocumentSerializationTest.java
+++ b/connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/document/BlankObjectMapperDocumentSerializationTest.java
@@ -1,0 +1,87 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.core.document;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.document.jackson.DocumentReferenceModel.CamundaDocumentMetadataModel;
+import io.camunda.connector.document.jackson.DocumentReferenceModel.CamundaDocumentReferenceModel;
+import io.camunda.connector.feel.jackson.FeelContextAwareObjectReader;
+import io.camunda.connector.feel.jackson.JacksonModuleFeelFunction;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Regression test for GitHub issue #6946.
+ *
+ * <p>When a FEEL expression evaluates against a context that contains a {@link CamundaDocument},
+ * the FEEL deserializer ends up calling {@code AbstractFeelDeserializer.BLANK_OBJECT_MAPPER
+ * .valueToTree(contextMap)}. The bug is that on 8.6/8.7/8.8 {@code BLANK_OBJECT_MAPPER} is a vanilla
+ * {@code new ObjectMapper()} — {@code FAIL_ON_EMPTY_BEANS=true} and no document serializer
+ * registered — so {@code valueToTree} throws {@link
+ * com.fasterxml.jackson.databind.exc.InvalidDefinitionException} for any {@link CamundaDocument}.
+ *
+ * <p>This test reproduces the user-facing failure path through the public API
+ * ({@link FeelContextAwareObjectReader}) and asserts that no exception is thrown.
+ *
+ * <p><b>Expected behaviour:</b>
+ *
+ * <ul>
+ *   <li>On stable/8.8 (and 8.6, 8.7) the test <b>fails</b> — {@code BLANK_OBJECT_MAPPER = new
+ *       ObjectMapper()} surfaces the bug as an exception.
+ *   <li>On main the test <b>passes</b> — {@code BLANK_OBJECT_MAPPER =
+ *       ConnectorsObjectMapperSupplier.getCopy()} has {@code FAIL_ON_EMPTY_BEANS=false}, so the
+ *       call no longer throws (though the document is silently emptied — see #6946 follow-up).
+ * </ul>
+ *
+ * <p>A complete fix would also register {@code JacksonModuleDocumentSerializer} on the blank mapper
+ * so the document reference is preserved rather than serialized as {@code {}}.
+ */
+class BlankObjectMapperDocumentSerializationTest {
+
+  private record TargetType(Supplier<String> value) {}
+
+  @Test
+  void feelEvaluationWithDocumentInContextDoesNotThrow() {
+    ObjectMapper mapper = new ObjectMapper().registerModule(new JacksonModuleFeelFunction());
+
+    var metadata = new CamundaDocumentMetadataModel(null, null, null, null, null, null, null);
+    var reference = new CamundaDocumentReferenceModel("store-1", "doc-1", "hash-1", metadata);
+    // Real CamundaDocument, not a Mockito mock — Mockito's proxy exposes internal getters that
+    // make the bean non-empty and accidentally bypass the FAIL_ON_EMPTY_BEANS path we want to
+    // exercise. The store is unused here because no content is read.
+    Map<String, Object> feelContext =
+        Map.of("doc", new CamundaDocument(metadata, reference, null));
+
+    // The FEEL expression itself is trivial — the failure is triggered by serializing the
+    // context map (which contains a CamundaDocument) into a JsonNode for the FEEL engine,
+    // not by evaluating the expression.
+    String json = "{ \"value\": \"= \\\"hello\\\"\" }";
+
+    assertThatNoException()
+        .as(
+            "Issue #6946: FEEL deserialization must not fail when the context contains a "
+                + "CamundaDocument. This test fails on stable/8.8 and below, and passes on main.")
+        .isThrownBy(
+            () ->
+                FeelContextAwareObjectReader.of(mapper)
+                    .withStaticContext(feelContext)
+                    .readValue(json, TargetType.class));
+  }
+}

--- a/connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/document/BlankObjectMapperDocumentSerializationTest.java
+++ b/connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/document/BlankObjectMapperDocumentSerializationTest.java
@@ -32,13 +32,13 @@ import org.junit.jupiter.api.Test;
  *
  * <p>When a FEEL expression evaluates against a context that contains a {@link CamundaDocument},
  * the FEEL deserializer ends up calling {@code AbstractFeelDeserializer.BLANK_OBJECT_MAPPER
- * .valueToTree(contextMap)}. The bug is that on 8.6/8.7/8.8 {@code BLANK_OBJECT_MAPPER} is a vanilla
- * {@code new ObjectMapper()} — {@code FAIL_ON_EMPTY_BEANS=true} and no document serializer
+ * .valueToTree(contextMap)}. The bug is that on 8.6/8.7/8.8 {@code BLANK_OBJECT_MAPPER} is a
+ * vanilla {@code new ObjectMapper()} — {@code FAIL_ON_EMPTY_BEANS=true} and no document serializer
  * registered — so {@code valueToTree} throws {@link
  * com.fasterxml.jackson.databind.exc.InvalidDefinitionException} for any {@link CamundaDocument}.
  *
- * <p>This test reproduces the user-facing failure path through the public API
- * ({@link FeelContextAwareObjectReader}) and asserts that no exception is thrown.
+ * <p>This test reproduces the user-facing failure path through the public API ({@link
+ * FeelContextAwareObjectReader}) and asserts that no exception is thrown.
  *
  * <p><b>Expected behaviour:</b>
  *
@@ -66,8 +66,7 @@ class BlankObjectMapperDocumentSerializationTest {
     // Real CamundaDocument, not a Mockito mock — Mockito's proxy exposes internal getters that
     // make the bean non-empty and accidentally bypass the FAIL_ON_EMPTY_BEANS path we want to
     // exercise. The store is unused here because no content is read.
-    Map<String, Object> feelContext =
-        Map.of("doc", new CamundaDocument(metadata, reference, null));
+    Map<String, Object> feelContext = Map.of("doc", new CamundaDocument(metadata, reference, null));
 
     // The FEEL expression itself is trivial — the failure is triggered by serializing the
     // context map (which contains a CamundaDocument) into a JsonNode for the FEEL engine,

--- a/connector-runtime/jackson-datatype-feel/pom.xml
+++ b/connector-runtime/jackson-datatype-feel/pom.xml
@@ -17,6 +17,10 @@
       <groupId>io.camunda.connector</groupId>
       <artifactId>connector-feel-wrapper</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.camunda.connector</groupId>
+      <artifactId>jackson-datatype-document</artifactId>
+    </dependency>
 
     <dependency>
       <groupId>org.junit.jupiter</groupId>

--- a/connector-runtime/jackson-datatype-feel/src/main/java/io/camunda/connector/feel/jackson/AbstractFeelDeserializer.java
+++ b/connector-runtime/jackson-datatype-feel/src/main/java/io/camunda/connector/feel/jackson/AbstractFeelDeserializer.java
@@ -20,8 +20,10 @@ import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
 import com.fasterxml.jackson.databind.deser.ContextualDeserializer;
 import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import io.camunda.connector.document.jackson.JacksonModuleDocumentSerializer;
 import io.camunda.connector.feel.FeelEngineWrapper;
 import java.io.IOException;
 import java.util.function.Supplier;
@@ -39,8 +41,16 @@ public abstract class AbstractFeelDeserializer<T> extends StdDeserializer<T>
    * aware of any registered modules. It should not be used to deserialize the final result. For
    * final results, use the {@link DeserializationContext} object passed to {@link
    * #doDeserialize(JsonNode, JsonNode, DeserializationContext)} instead.
+   *
+   * <p>{@link SerializationFeature#FAIL_ON_EMPTY_BEANS} is disabled and {@link
+   * JacksonModuleDocumentSerializer} is registered so that FEEL contexts containing a {@code
+   * CamundaDocument} can be serialized into a tree without throwing — and so the document
+   * reference is preserved instead of being silently emptied. See issue #6946.
    */
-  protected static final ObjectMapper BLANK_OBJECT_MAPPER = new ObjectMapper();
+  protected static final ObjectMapper BLANK_OBJECT_MAPPER =
+      new ObjectMapper()
+          .disable(SerializationFeature.FAIL_ON_EMPTY_BEANS)
+          .registerModule(new JacksonModuleDocumentSerializer());
 
   protected AbstractFeelDeserializer(FeelEngineWrapper feelEngineWrapper, boolean relaxed) {
     super(String.class);

--- a/connector-runtime/jackson-datatype-feel/src/main/java/io/camunda/connector/feel/jackson/AbstractFeelDeserializer.java
+++ b/connector-runtime/jackson-datatype-feel/src/main/java/io/camunda/connector/feel/jackson/AbstractFeelDeserializer.java
@@ -44,8 +44,8 @@ public abstract class AbstractFeelDeserializer<T> extends StdDeserializer<T>
    *
    * <p>{@link SerializationFeature#FAIL_ON_EMPTY_BEANS} is disabled and {@link
    * JacksonModuleDocumentSerializer} is registered so that FEEL contexts containing a {@code
-   * CamundaDocument} can be serialized into a tree without throwing — and so the document
-   * reference is preserved instead of being silently emptied. See issue #6946.
+   * CamundaDocument} can be serialized into a tree without throwing — and so the document reference
+   * is preserved instead of being silently emptied. See issue #6946.
    */
   protected static final ObjectMapper BLANK_OBJECT_MAPPER =
       new ObjectMapper()

